lib/virtfs/ext3/directory.rb

```diff
@@ -1,51 +1,85 @@
-require 'fs/ext3/file_data'
-require 'fs/ext3/directory_entry'
+require_relative 'file_data'
+require_relative 'directory_entry'
 
-module Ext3
+module VirtFS::Ext3
   class Directory
     ROOT_DIRECTORY = 2
 
-    def initialize(sb, inodeNum = ROOT_DIRECTORY)
-      raise "Ext3::Directory.initialize: Nil superblock"   if sb.nil?
-      raise "Ext3::Directory.initialize: Nil inode number" if inodeNum.nil?
-      @sb = sb; @inodeNum = inodeNum
-      @inodeObj = sb.getInode(inodeNum)
-      @data = FileData.new(@inodeObj, @sb).read
+    attr_accessor :fs
+
+    def initialize(fs, sb, inode_num = ROOT_DIRECTORY)
+      raise "nil superblock"   if sb.nil?
+      raise "nil inode number" if inode_num.nil?
+      @fs        = fs
+      @sb        = sb
+      @inode_num = inode_num
+      @inode_obj = sb.get_inode(inode_num)
+      @data      = sb.ext4? ? @inode_obj.read : FileData.new(@inode_obj, @sb).read
+    end
```

Comment:
@jerryk55 the Ext4::Inode class provides a 'read' method which we can use to access the inode data directly, but not such method exists in the ext3 module. This was added so that both modules can leverage the common Directory code.

At some point the "read" method should be added to the Ext3::inode class for consistency.